patches/tModLoader/Terraria/DataStructures/PlayerDrawSet.cs.patch

```diff
@@ -0,0 +1,46 @@
+--- src/Terraria/Terraria/DataStructures/PlayerDrawSet.cs
++++ src/tModLoader/Terraria/DataStructures/PlayerDrawSet.cs
+@@ -10,6 +_,14 @@
+ {
+ 	public struct PlayerDrawSet
+ 	{
++		//A sad trick to avoid big patches for disabling lighting below.
++		private class Lighting
++		{
++			public static bool use;
++			public static Color GetColor(int x, int y) => use ? Terraria.Lighting.GetColor(x, y) : Color.White;
++			public static Color GetColor(int x, int y, Color oldColor) => use ? Terraria.Lighting.GetColor(x, y, oldColor) : oldColor;
++		}
++
+ 		public List<DrawData> DrawDataCache;
+ 		public List<int> DustCache;
+ 		public List<int> GoreCache;
+@@ -118,7 +_,9 @@
+ 
+ 		public Vector2 Center => new Vector2(Position.X + (float)(drawPlayer.width / 2), Position.Y + (float)(drawPlayer.height / 2));
+ 
+-		public void BoringSetup(Player player, List<DrawData> drawData, List<int> dust, List<int> gore, Vector2 drawPosition, float shadowOpacity, float rotation, Vector2 rotationOrigin) {
++		public void BoringSetup(Player player, List<DrawData> drawData, List<int> dust, List<int> gore, Vector2 drawPosition, float shadowOpacity, float rotation, Vector2 rotationOrigin, bool useLighting = true) {
++			Lighting.use = useLighting;
```

Comment:
could reduce a couple lines of patches/maintenance by setting this at the call site

Reply:
This is public though, and is usable by modders.

Reply:
Can you give me a use case? it's only called in `DrawPlayer` as far as I can see

Reply:
It could be useful if someone wants to implement their own IPlayerRenderer, I guess.

Reply:
in that case they could also set `Lighting.use` to true, if you make the field public

Reply:
(Resolved in discord.)